src/pages/index.js

```diff
@@ -30,55 +36,198 @@ const addCardPopup = new PopupWithForm(
-  const userData = userInfo.getUserInfo();
-  profileTitleInput.value = userData.name;
-  profileDescriptionInput.value = userData.about;
-  profileEditPopup.open();
-});
-
-addCardButton.addEventListener("click", () => addCardPopup.open());
+// Constants
+const profileEditBtn = document.querySelector("#profile__edit-button");
+const addCardButton = document.querySelector(".profile__add-button");
+const profileTitleInput = document.querySelector("#profile__name-input");
+const profileDescriptionInput = document.querySelector(
+  "#profile__subheading-input"
+);
+const trashModalSubmitBtn = document.querySelector(".modal__button-trash");
+const editProfileImage = document.querySelector(".profile__avatar-edit");
 
-// Event handlers
+// Functions
 function handleProfileEditSubmit(profileData) {
-  const { modal__input_type_name: name, modal__input_type_description: about } =
-    profileData;
-  userInfo.setUserInfo(name, about);
-  profileEditPopup.close();
+  const name = profileData.modal__input_type_name;
+  const about = profileData.modal__input_type_description;
+
+  profileEditPopup.renderLoading(true);
+
+  api
+    .patchProfileInfo(name, about)
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+    });
 }
 
 function handleAddCardSubmit(newCardData) {
-  renderer(newCardData);
-  addCardPopup.close();
+  addCardPopup.renderLoading(true);
+
+  api
+    .postCards(newCardData)
+    .then((cardData) => {
+      renderer(cardData);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+}
+
+function handleAvatarSubmit(formData) {
+  const avatarLink = formData["modal__form-input-link"];
+
+  editAvatarPopup.renderLoading(true);
+
+  api
+    .patchProfileAvatar(avatarLink)
+    .then(() => {
+      userInfo.setAvatarPic(avatarLink);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoading(false);
+    });
+}
+
+function handleImageClick(card) {
+  previewImagePopup.open({ name: card.name, link: card.link });
+}
+
+function handleDelete(card) {
+  trashConfirmPopup.open();
+  trashModalSubmitBtn.addEventListener(
+    "click",
+    () => {
+      api
+        .deleteCard(card._id)
+        .then(() => {
+          card.removeCard();
+          trashConfirmPopup.close();
+        })
+        .catch((err) => {
+          console.error("Error deleting card:", err);
+        });
+    },
+    { once: true }
+  );
 }
 
-function handleImageClick(name, link) {
-  previewImagePopup.open({ name, link });
+function handleLikeIconClick(card) {
+  const apiAction = card._isLiked
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.